guix-science/packages/geant4.scm

```diff
@@ -0,0 +1,699 @@
+                                                  datadir "/share/geant4/data"))))
+       #:phases (modify-phases %standard-phases
+                  (add-after 'install 'install-data
+                    (lambda* (#:key inputs outputs #:allow-other-keys)
+                      (let ((G4NDL (assoc-ref inputs "G4NDL"))
+                            (G4EMLOW (assoc-ref inputs "G4EMLOW"))
+                            (G4PhotonEvaporation (assoc-ref inputs
+                                                  "G4PhotonEvaporation"))
+                            (G4RadioactiveDecay (assoc-ref inputs
+                                                 "G4RadioactiveDecay"))
+                            (G4PARTICLEXS (assoc-ref inputs "G4PARTICLEXS"))
+                            (G4PII (assoc-ref inputs "G4PII"))
+                            (G4RealSurface (assoc-ref inputs "G4RealSurface"))
+                            (G4SAIDDATA (assoc-ref inputs "G4SAIDDATA"))
+                            (G4ABLA (assoc-ref inputs "G4ABLA"))
+                            (G4INCL (assoc-ref inputs "G4INCL"))
+                            (G4ENSDFSTATE (assoc-ref inputs "G4ENSDFSTATE"))
+                            (G4TENDL (assoc-ref inputs "G4TENDL"))
+                            (datadir (string-append (assoc-ref outputs "out")
+                                                    "/share/geant4/data")))
+                        (display (list "Data archives:"
+                                       G4NDL
+                                       G4EMLOW
+                                       G4PhotonEvaporation
+                                       G4RadioactiveDecay
+                                       G4PARTICLEXS
+                                       G4PII
+                                       G4RealSurface
+                                       G4SAIDDATA
+                                       G4ABLA
+                                       G4INCL
+                                       G4ENSDFSTATE))
+                        (newline)
+                        (mkdir-p datadir)
+                        (invoke "tar" "xvf" G4NDL "-C" datadir)
+                        (invoke "tar" "xvf" G4EMLOW "-C" datadir)
+                        (invoke "tar" "xvf" G4PhotonEvaporation "-C" datadir)
+                        (invoke "tar" "xvf" G4RadioactiveDecay "-C" datadir)
+                        (invoke "tar" "xvf" G4PARTICLEXS "-C" datadir)
+                        (invoke "tar" "xvf" G4PII "-C" datadir)
+                        (invoke "tar" "xvf" G4RealSurface "-C" datadir)
+                        (invoke "tar" "xvf" G4SAIDDATA "-C" datadir)
+                        (invoke "tar" "xvf" G4ABLA "-C" datadir)
+                        (invoke "tar" "xvf" G4INCL "-C" datadir)
+                        (invoke "tar" "xvf" G4ENSDFSTATE "-C" datadir)
+                        (invoke "tar" "xvf" G4TENDL "-C" datadir)))))
```

Comment:
This looks all very much like the package above.  Can we inherit this stuff as well?

Reply:
Use `substitute-keyword-arguments` to keep some keyword arguments unchanged and only override those that have changed.